src/Avalonia.Base/Styling/Activators/ScreenActivator.cs

```diff
@@ -0,0 +1,28 @@
+namespace Avalonia.Styling.Activators
+{
+    internal sealed class WidthActivator : ContainerQueryActivatorBase
+    {
+        private readonly (QueryComparisonOperator @operator, double value) _argument;
+
+        public WidthActivator(Visual visual, (QueryComparisonOperator @operator, double value) argument, string? containerName = null) : base(visual, containerName)
+        {
+            _argument = argument;
+        }
+
+        protected override bool EvaluateIsActive() => (CurrentContainer?.ContainerType == Layout.ContainerType.Width || CurrentContainer?.ContainerType == Layout.ContainerType.WidthAndHeight) 
```

Comment:
What happens if container type changes at runtime?

Reply:
It wouldn't evaluate as long as it isn't a container type that applies to the dimension.